includes/classes/shopping_cart.php

```diff
@@ -69,7 +69,7 @@ class shoppingCart extends base {
    * @var string
    */
   var $display_debug_messages = FALSE;
-  var $flag_duplicate_msgs_set = FALSE;
+  var $flag_duplicate_msgs_set = array(); 
```

Comment:
Where is this value used as an array? Current processing uses TRUE/FALSE other than these two changes.

Reply:
You're right - this was carried over from 1.5.6 where the logic is different.